test/e2e/service_test.go

```diff
@@ -62,6 +68,36 @@ func (test *e2eTest) serviceCreateDuplicate(t *testing.T, serviceName string) {
 	assert.ErrorContains(t, err, "the service already exists")
 }
 
+func (test *e2eTest) serviceCreatePrivate(t *testing.T, serviceName string) {
+	out, err := test.kn.RunWithOpts([]string{"service", "list", serviceName}, runOpts{NoNamespace: false})
+	assert.NilError(t, err)
+	assert.Check(t, strings.Contains(out, serviceName), "The service does not exist yet")
+
+	_, err = test.kn.RunWithOpts([]string{"service", "create", serviceName,
+		"--image", KnDefaultTestImage, "--cluster-local"}, runOpts{NoNamespace: false, AllowError: true})
+	assert.NilError(t, err)
+
+	assert.Check(t, util.ContainsAllIgnoreCase(out, "service", serviceName, "creating", "namespace", test.kn.namespace, "ready"))
```

Comment:
I think we should read and verify the labels applied for private service here.

Reply:
And check for removal. Good call. Adding that.